src/apiTest/kotlin/com/ampnet/auditornode/script/api/classes/HttpClientJavaScriptApiTest.kt

```diff
@@ -0,0 +1,868 @@
+package com.ampnet.auditornode.script.api.classes
+
+import arrow.core.right
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.ampnet.auditornode.ApiTestBase
+import com.ampnet.auditornode.jsAssertions
+import com.ampnet.auditornode.model.error.EvaluationError.InvalidInputValueError
+import com.ampnet.auditornode.script.api.model.AuditResult
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.equalTo
+import com.github.tomakehurst.wiremock.client.WireMock.get
+import com.github.tomakehurst.wiremock.client.WireMock.post
+import com.github.tomakehurst.wiremock.client.WireMock.request
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
+import com.github.tomakehurst.wiremock.http.HttpHeader
+import com.github.tomakehurst.wiremock.http.HttpHeaders
+import io.micronaut.http.HttpRequest
+import io.micronaut.http.MediaType
+import io.micronaut.http.cookie.Cookie
+import io.micronaut.http.cookie.SameSite
+import io.micronaut.http.simple.cookies.SimpleCookie
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest
+import org.intellij.lang.annotations.Language
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+@MicronautTest
+class HttpClientJavaScriptApiTest : ApiTestBase() {
+
+    private val wireMockServer = WireMockServer(wireMockConfig().dynamicPort())
+
+    @BeforeEach
+    fun beforeEach() {
+        wireMockServer.start()
+    }
+
+    @AfterEach
+    fun afterEach() {
+        wireMockServer.stop()
+    }
```

Comment:
Is it possible to restart/reset/invalidate the server after each test?

Reply:
I mean this is what `start` and `stop` already do... There is a `@WireMockRule` which would do this automatically for us, but it does not work with JUnit 5. I could maybe do this, but I don't think there is much benefit:

```kotlin
@BeforeAll
fun beforeAll() {
    wireMockServer.start()
}

@AfterEach
fun afterEach() {
    wireMockServer.resetAll()
}

@AfterAll
fun afterAll() {
    wireMockServer.stop()
}
```

WireMock server is already very quick to start/stop, so doing it like this would not give us any noticeable performance in tests.